shared/js/content-scripts/block-cookie.js

```diff
@@ -0,0 +1,92 @@
+// Set up 3rd party cookie blocker
+(function cookieBlocker () {
+    function getFrameUrl () {
+        let url = document.location.href
+        let parentFrame = (document != window.top) && window.parent
+        while (parentFrame && url && !url.startsWith('http')) {
+            try {
+                url = parentFrame.document.location.href
+            } catch (ex) {
+                // ignore 'Blocked a fr[ame with origin "..."
+                // from accessing a cro]ss-origin frame.' exceptions
+            }
+            parentFrame = (parentFrame != window.top) && parentFrame.parent
+        }
+        return url
+    }
+
+    if (window.top == window) {
+        return
+    }
+
+    chrome.runtime.sendMessage({
+        'checkThirdParty': true,
+        frameUrl: getFrameUrl()
+    }, function (isThridParty) {
+        if (isThridParty) {
+            const scriptString = `
```

Comment:
can we move this code in huge strings to a function like:

```js
function injectedClearCookies() {
   // Clear previously set cookies
  …
}

…

const scriptString = `(function() {${injectedClearCookies.toString()}()})()`

```

This way syntax will be linted and higlighted in diffs 🌈

Reply:
This is magical ✨